src/cpu/flexcpu/simple_dataflow_thread.hh

```diff
@@ -433,15 +433,15 @@ class SDCPUThread : public ThreadContext
      *
      * @param inst_ptr The instruction for which we want to track uses.
      */
-    void populateUses(std::shared_ptr<InflightInst> inst_ptr);
+    void populateUses(std::shared_ptr<InflightInst>& inst_ptr);
 
     /**
      * This function makes a request for the branch predictor for the given
      * control instruction.
      *
      * @param inst_ptr The control instruction which needs a prediction.
      */
-    void predictCtrlInst(std::shared_ptr<InflightInst> inst_ptr);
+    void predictCtrlInst(std::shared_ptr<InflightInst>& inst_ptr);
```

Comment:
Pretty sure this can actually be `const std::shared_ptr<InflightInst>& inst_ptr`. Since it's the pointer that's const, not the InflightInst it references. If it were `std::shared_ptr<const InflightInst>& inst_ptr` that would be a different story.